examples/tf/petting_zoo/atari/pong/recurrent/decentralised/run_madqn.py

```diff
@@ -87,7 +87,6 @@ def main(_: Any) -> None:
         exploration_scheduler_fn=LinearExplorationScheduler(
             epsilon_start=1.0, epsilon_min=0.05, epsilon_decay=1e-6
         ),
-        shared_weights=False,
```

Comment:
If we want to use shared weights here, can we rather just make this explicitly `True` rather than not setting the argument? Seems clearer to me.

Reply:
I removed this statement as it does not make sense to share weights in multi-agent of Pong since there are two agent types in a competitive game (It is possible to make it a single-agent problem). The new functionality makes sure that each agent has there own network even when shared weights is True since they are different types But I guess you are right in that it is better to be explicit.

Reply:
@arnupretorius @DriesSmit Would it not make more sense to have shared_weights set to false by default?

Reply:
Sharing weights is more common in MARL than having separate parameters for each agent, or at least in the cooperative, mixed (team) settings. I think it makes sense to have `shared_weights` default to `True`.